asset/asset_service_impl.go

```
@@ -0,0 +1,312 @@
+	}, nil
+}
+
+func (s *assetService) handleArchiveExtraction(ctx context.Context, assetID AssetID, storageKey string) {
+	err := s.assetRepo.UpdateExtractionStatus(ctx, assetID, ExtractionStatusInProgress)
+	if err != nil {
+		return
+	}
+
+	reader, err := s.storage.Get(ctx, storageKey)
+	if err != nil {
+		s.assetRepo.UpdateExtractionStatus(ctx, assetID, ExtractionStatusFailed)
```

golangci-lint (1.64.8)

Error return value of `s.assetRepo.UpdateExtractionStatus` is not checked

(errcheck)